Optional input and adjusted defaults

consideRatio opened the pull request:
I think the k3s-version and helm-version defaults really should be blank and therefore become the latest k3s / helm version, but I also opined on the other flags where I'm more open to leaving at it is if you think so @manics.

I would argue for having metrics/traefik-enabled flags to be false by default as they add some weight to the installation, which I think can merit an opt-in rather than opt-out.

consideRatio commented:
![image](https://user-images.githubusercontent.com/3837114/98457465-c9826680-2187-11eb-8224-012f501d838b.png)

I added a commit also to quote the environment variables, so that if they are a blank string, the error above would not happen in this scenario:

```bash
# this will be fine as it seems like the == operator got a blank string to compare with "something"
if [[ "$CAN_BE_BLANK" == "something" ]]; then echo ok; fi

# this will fail fine as it seems like the == operator got nothing at all to compare with "something"
if [[ $CAN_NOT_BE_BLANK == "something" ]]; then echo ok; fi
```

manics commented:
Thanks, defaulting to latest is great- I copied your original code and didn't realise you could leave the default unset :smile:. Any chance you could add a test scenario, presumably just add `k3s: ""`, `helm: ""` to the testing matrix?

Not sure about changing the defaults though. I decided to enabled local storage which is disabled in the original Z2JH CI because I think it's generally useful, and opens up more testing options both for Z2JH and other Helm charts that can use this. Similarly for ingress, it's not currently used in Z2JH (though it could be as one of the test scenarios) but can be generally useful for other charts. That leave metrics which I agree is not very useful in CI, but it does make it slightly harder for user to reason about what flags are chosen, which is why I'm more inclined to leave flags as their defaults except where it conflicts with something (e.g. to enable the Calico network controller).

consideRatio commented:
@manics deal!
- Reverted the default values changes for metrics/traefik but retained the new blank-string defaults for k3s/helm version.
- Added a test where we test to explicitly specify k3s/helm-version to a blank string.
- Fixed a bug i introduced where a comparison like `[[ "v1.16.15+k8s" == "v1.16.*" ]]` would fail because it became a string literal comparison by adding the quotes.

manics reviewed (approved):
Thanks!

consideRatio commented:
@manics thanks for your work on this and your review! :tada:
